CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ option(IREE_BUILD_TESTS "Builds IREE unit tests." ON)
 option(IREE_BUILD_SAMPLES "Builds IREE sample projects." ON)
 option(IREE_BUILD_DEBUGGER "Builds the IREE debugger app." OFF)
 
+if(${IREE_BUILD_SAMPLES})
+  set(IREE_BUILD_COMPILER ON CACHE BOOL "Build the IREE compiler for sample projects." FORCE)
```

Comment:
Do all of the samples need the compiler? I'd rather not set `IREE_BUILD_COMPILER` on by default as long as TensorFlow and Python are required, since that adds friction to getting started with the rest of the project.

How about either:
  1) Keeping this, but flipping `IREE_BUILD_SAMPLES` to OFF by default
  2) Removing this, and conditionally enabling each sample based on `IREE_BUILD_SAMPLES` and `IREE_BUILD_COMPILER` together (as needed)

Reply:
`iree-tblgen` and `iree-translate` are only build if `IREE_BUILD_COMPILER` is set
https://github.com/google/iree/blob/14866cac626bbdadfd409e6ce7011056dd4c35c8/iree/tools/CMakeLists.txt#L15
as they depend on several `iree::compiler` components. At least the hal and the rt samples require `iree-tblgen` and `iree-translate`. To be honest, did check the freshly landed custom
_module samples.

Reply:
`iree::samples::custom_modules` relies on`iree_bytecode_module` as well. Therefore, I would go with option 1.

Reply:
Okay, I chatted with @benvanik offline about this and we'd actually like to keep what you originally had, but also flip `IREE_BUILD_COMPILER` to ON by default as well. We can separately change what the CI is building: https://github.com/google/iree/blob/14866cac626bbdadfd409e6ce7011056dd4c35c8/.github/workflows/ci.yml#L88-L92

The CMake build doesn't include TensorFlow and Python yet (unlike Bazel), and when it does we can add something like `IREE_BUILD_TENSORFLOW`. The compiler and samples should really have a stable build, and enabling them by default (now that they work :D!) will help there.

Separately, I (or someone else) can make the Vulkan samples conditionally enabled, so people building the project for CPU execution don't need to go through that setup.

Reply:
There we go :)

I am still not sure we can build the TensorFlow parts required by Python with CMake, but for the remaining parts we should.

Introducing a build switch for the Vulkan samples (also the runtime and compiler parts?) sounds like a reasonable idea. Maybe `IREE_ENABLE_VULKAN` build switch in the main CMakeLists.txt?